File: comp-228-java-programming/week04/src/thiskeyword/Employee.java
package thiskeyword;

public class Employee {
    private int id;
    private String name;

    public Employee(){
        this(1111, "Mark");
    }



    public Employee(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public void setAllValues(int id, String name) {
        this.id = id;
        this.name = name;
    }

    @Override
    public String toString() {
        return "Employee{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: comp-212-programming-3/01 - Generics, Type Constraints, and Ext Methods/MaximumTest/MaximumTest/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MaximumTest
{
    public class Employee  : IComparable<Employee>  
    {
        public Employee(string name, decimal salary)
        {
            this.Name = name;
            this.Salary = salary;
        }

        public string Name { get; }
        public decimal Salary { get; private set; }

        public override string ToString() => $"{Name}, {Salary:C}";

        public static bool CompareSalary(Employee e1, Employee e2) =>
          e1.Salary < e2.Salary;

        public int CompareTo(Employee e)
        {
            if (Salary < e.Salary) return -1;
            else if (Salary == e.Salary) return 0;
                    return 1;
        }
    }
}
